server/models/base/ArchivableModel.ts

```diff
@@ -0,0 +1,25 @@
+/* eslint-disable @typescript-eslint/ban-types */
+import { AllowNull, Column, IsDate } from "sequelize-typescript";
+import ParanoidModel from "./ParanoidModel";
+
+class ArchivableModel<
+  TModelAttributes extends {} = any,
+  TCreationAttributes extends {} = TModelAttributes
+> extends ParanoidModel<TModelAttributes, TCreationAttributes> {
```

Comment:
Extending from `ParanoidModel` here. There's a limitation here that a non-paranoid model which may be archivable, can't really extend from `ArchivableModel`. `Notification` model is one such example.

It doesn't seem trivial because Typescript doesn't provide multiple inheritance out of the box. Haven't looked into this yet but seems like mixins can help achieve this. Could be scoped out separately if we want to explore that?